app/views/admin/contents/_fieldset.html.haml

```diff
@@ -2,7 +2,11 @@
   %legend{align: "center"}= name
   - preferences.each do |key|
     - type = ContentConfig.preference_type(key)
+    - if name == t('admin.contents.edit.main_links') && type == :boolean
```

Comment:
Having a branch in a loop is a bit ugly. I'm also struggling to understand it. So the translation here has to match the translation in the ContentsController. And then you have a different label depending on the type and if it's a special menu item checkbox.

It does look really nice on your screenshot. I'm just wondering if we can simplify this. But that would probably involve having our own model for preferences which decorates the Spree preferences. Probably out of scope of this pull request.

Reply:
Yes! This isn't nice but it's the best I could do without creating a new configuration screen, new preferences, etc...
this IF reads: "if it's a checkbox in the content main links, make the checkbox text be the title and the url merged"